syntax = "proto3";

package beneficiary;

import "error/error.proto";

message Beneficiary {
    string id = 1;
    string beneficiary_name = 2;
    string email = 3;
    string website_url = 4;
    string phone_number = 5;
    string location_city = 6;
    string location_state = 7;
    string ein = 8;
    string beneficiary_description = 9;
    string beneficiary_size = 10;
}

message GetBeneficiaryRequest {
    string beneficiary_id = 1;
}

message GetBeneficiaryResponse {
    Beneficiary beneficiary = 1;
    repeated error.Error errors = 2;
}

message CreateBeneficiaryRequest {
    string beneficiary_name = 1;
    string email = 2;
    string website_url = 3;
    string phone_number = 4;
    string location_city = 5;
    string location_state = 6;
    string ein = 7;
    string beneficiary_description = 8;
    string beneficiary_size = 9;
    string user_email = 10;
}

message CreateBeneficiaryResponse {
    Beneficiary beneficiary = 1;
    repeated error.Error errors = 2;
}

service BeneficiaryService {
    rpc GetBeneficiary(GetBeneficiaryRequest) returns (GetBeneficiaryResponse);
    rpc CreateBeneficiary(CreateBeneficiaryRequest) returns (CreateBeneficiaryResponse);
}
